feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/staking/start/setupStakingType/EditingStakingTypeSelectionMixin.kt

```diff
@@ -0,0 +1,111 @@
+package io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType
+
+import io.novafoundation.nova.common.utils.combine
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.RecommendableMultiStakingSelection
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.SelectionTypeSource
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.store.StartMultiStakingSelectionStoreProvider
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.selection.store.currentSelectionFlow
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.types.CompoundStakingTypeDetailsProvidersFactory
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.common.types.StakingTypeDetailsProvider
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType.direct.EditingStakingTypeValidationSystem
+import io.novafoundation.nova.feature_staking_impl.domain.staking.start.setupStakingType.model.EditableStakingType
+import io.novafoundation.nova.runtime.ext.StakingTypeGroup
+import io.novafoundation.nova.runtime.ext.group
+import io.novafoundation.nova.runtime.multiNetwork.ChainWithAsset
+import io.novafoundation.nova.runtime.multiNetwork.chain.model.Chain
+import java.math.BigInteger
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.map
+
+class EditingStakingTypeSelectionMixinFactory(
+    private val currentSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+    private val editableSelectionStoreProvider: StartMultiStakingSelectionStoreProvider,
+    private val compoundStakingTypeDetailsProvidersFactory: CompoundStakingTypeDetailsProvidersFactory
+) {
+
+    suspend fun create(
+        scope: CoroutineScope,
+        chainWithAsset: ChainWithAsset,
+        availableStakingTypes: List<Chain.Asset.StakingType>
+    ): EditingStakingTypeSelectionMixin {
+        return EditingStakingTypeSelectionMixin(
+            currentSelectionStoreProvider,
+            editableSelectionStoreProvider,
+            compoundStakingTypeDetailsProvidersFactory.create(scope, chainWithAsset, availableStakingTypes),
+            scope
+        )
+    }
+}
```

Comment:
Why did you decide to extract it into a separate mixin? We dont plan to reuse it anywhere so this logic can be part of viewModel directly

Reply:
I just wanted to take out part of the logic from the view model for convenience

Reply:
I dont think it give us any profits here since 
1. This logic wont make ViewModel big
2. In ViewModel you are stil flatMapping over fields of this mixin so ViewModel still have part of the logic